### PR TITLE
Add Android builds optionally to unified build

### PR DESCRIPTION
**Pull request by mspang**
To enable, specify android_ndk_root & enable_android_builds=true.

### CODE REVIEW

**Review by saurabhst — APPROVED** [review]

File: .github/workflows/android.yaml [hunk 1]
```diff
@@ -24,10 +24,10 @@ jobs:
 
         strategy:
             matrix:
-                type: [arm, arm64]
+                type: [arm, arm64, x64]
 
         env:
-            BUILD_TYPE: ${{ matrix.type }}
+            BUILD_TYPE: android_${{ matrix.type }}
```
Comment (rwalker-apple) [inline .github/workflows/android.yaml:30]: just curious, where does BUILD_TYPE android_XXX get used?  looks like you removed the only reference to it below on line 48

File: gn/chip/tools.gni [hunk 2]
```diff
@@ -14,5 +14,5 @@
 
 declare_args() {
   # Build CHIP tools.
-  chip_build_tools = current_os != "freertos"
+  chip_build_tools = current_os != "freertos" && current_os != "android"
```
Comment (rwalker-apple) [inline gn/chip/tools.gni:17]: this could as well be accomplished with an arg, as darwin builds do

**Review by rwalker-apple — APPROVED** [review]

File: .github/workflows/android.yaml (hunk 1 above)
Comment (mspang) [reply .github/workflows/android.yaml:30]: The helper script sets the out dir name from this.

File: gn/chip/tools.gni (hunk 2 above)
Comment (rwalker-apple) [reply gn/chip/tools.gni:17]: what are your thoughts, @mspang

Comment (mspang) [reply gn/chip/tools.gni:17]: This is just a default value, you can still set it via args. Setting platform-specific defaults it make configuring on the command line easier, in most cases you only need to specify CPU & OS.

**Review by robszewczyk — APPROVED** [review]

**Pull request merged**